src/ol/style/fill.js

```diff
@@ -31,6 +31,18 @@ ol.style.Fill = function(opt_options) {
 
 
 /**
+ * Clones the style.
+ * @return {ol.style.Fill} The cloned style.
+ * @api
+ */
+ol.style.Fill.prototype.clone = function() {
+  return new ol.style.Fill({
+    color: (this.getColor() instanceof Array) ? this.getColor().slice(0) : this.getColor()
```

Comment:
@marcjansen already agreed on using `.slice()`. i just wanted to ask about skipping all the instanceof checks.